uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/API/Query/LeaveRequestSelect.php

```diff
@@ -217,4 +221,39 @@ private function addGroupBy(CRM_Utils_SQL_Select $query) {
     $query->groupBy(['a.id']);
   }
 
+  /**
+   * Returns if the balance_change field should be included on the returned results
+   *
+   * @return bool
+   */
+  private function shouldReturnBalanceChange() {
+    return $this->shouldReturnField('balance_change');
+  }
+
+  /**
+   * Returns if the dates field should be included on the returned results
+   *
+   * @return bool
+   */
+  private function shouldReturnDates() {
+    return $this->shouldReturnField('dates');
+  }
+
+  /**
+   * Returns, based on the "return" param, if the given field should be returned
+   * on the results.
+   *
+   * If "return" is empty, it means all the fields should be returned. Otherwise,
+   * a field will be returned only if "return" is not empty and it includes the
+   * field.
+   *
+   * @param string $field
+   *
+   * @return bool
+   */
+  private function shouldReturnField($field) {
+    return empty($this->params['return']) ||
+           (is_array($this->params['return']) && in_array($field, $this->params['return']));
```

Comment:
Why not `empty($this->params['return']) || $returnArrayHasField` ?
where `$returnArrayHasField = (is_array($this->params['return']) && in_array($field, $this->params['return']));`